contracts/utils/ERC721BatchTransfer.sol

```diff
@@ -0,0 +1,159 @@
+//SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+import { IERC721 } from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+
+/**
+ * @title ERC721 Batch Transfer
+ *
+ * @notice ETransfer ERC721 tokens in batches to a single wallet or multiple wallets. This supports ERC721M and ERC721CM contracts.
+ * @notice To use any of the methods in this contract the user has to approve this contract to control their tokens using either `setApproveForAll` or `approve` functions from the ERC721 contract.
+ */
+contract ERC721BatchTransfer {
+    error InvalidArguments();
+    error NotOwnerOfToken();
+    error InvalidCaller();
+
+    event BatchTransferToSingle(
+        address indexed contractAddress,
+        address indexed to,
+        uint256 amount
+    );
+
+    event BatchTransferToMultiple(
+        address indexed contractAddress,
+        uint256 amount
+    );
+
+    constructor() {}
+
+    modifier noZero() {
+        if (msg.sender == address(0)) revert InvalidCaller();
+        _;
+    }
+
+    /** 
+     * @notice Transfer multiple tokens to the same wallet using the ERC721.transferFrom method.
+     * @notice If you don't know what that means, use the `safeBatchTransferToSingleWallet` method instead
+     * @param erc721Contract the address of the nft contract
+     * @param to the address that will receive the nfts
+     * @param tokenIds the list of tokens that will be transferred
+     */
+    function batchTransferToSingleWallet(
+        IERC721 erc721Contract,
+        address to,
```

Comment:
should we revert if `to` is a null address?

Reply:
You mean `address(0)`? We should allow it so we can batch burn.